NF: Plannar projection of lines can be captured

maharshi-gor opened the pull request:
### NF: Plannar projection of lines can be captured

**PR Contents**

- Introduces a new actor called `line_projection`.
- Compute shader to calculate the intersection points.
- Fixes opacity issue in points material method.
- Adds functionality to update slices for groups apart from img slicer.
- Added Test cases.
- Added Tutorial for the same.

**NOTE: CAUTION**
I have freezed the pygfx version as it has created some breaking changes in the latest release. I have created an issue to track the same as well.

**Code to try**

Please use the tutorial `viz_line_projections.py` in the PR

**Screenshots**
<img width="389" height="727" alt="Screenshot 2025-09-16 at 12 10 28 AM" src="https://github.com/user-attachments/assets/d7254ced-9174-4852-a85f-706726a46411" />
<img width="660" height="685" alt="Screenshot 2025-09-16 at 12 11 53 AM" src="https://github.com/user-attachments/assets/a1b59f47-9e27-425a-83e3-5a0423ffa150" />

maharshi-gor committed:
NF: Plannar projection of lines can be captured

- Introduces a new actor called `line_projection`.
- Compute shader to calculate the intersection points.
- Fixes opacity issue in points material method.
- Adds functionality to update slices for groups apart from img slicer.
- Test cases for the functionality through out.
- Tutorial `viz_line_projections.py` introduced.
- Pygfx version freeze incorporate new changes.

Garyfallidis commented on fury/actor/planar.py:
```diff
@@ -692,3 +709,326 @@ def ring(
+
+    Parameters
+    ----------
+    lines : list
```
sequence

Garyfallidis commented on docs/examples/viz_line_projections.py:
```diff
@@ -0,0 +1,73 @@
+    colors=colors,
+    thickness=10,
+    outline_thickness=3,
+    outline_color=(0, 0, 0),
```
Default lift should be 0.

Garyfallidis commented:
See if you can support the latest version before merging this.

maharshi-gor committed:
Merge branch 'v2' of github.com:fury-gl/fury into streamline-slice

maharshi-gor committed:
RF: Comments Resolved.

- Changed the `line` to `sequence`.
- Updated the default lift to be `0`.

skoudoro commented on docs/examples/viz_line_projections.py:
```diff
@@ -0,0 +1,73 @@
+"""
+============================
```
too short, it generates a warning. 

I think I will activate warnings as errors in general. we start to accumulate warnings which is not good

maharshi-gor replied:
Let me create a fix PR for the usual test case warning we get and we can enable the warning to error option

skoudoro commented on docs/examples/viz_line_projections.py:
```diff
@@ -0,0 +1,73 @@
+
+projection = actor.line_projection(
+    lines,
+    plane=(0, 0, -1, 0),  # Plane equation: z = 0 or XY plane
```
This does not make sense to me, and it is not user friendly. 
Why not using directly `plane`  which is `actor.square`. Please, explain the link between both...

maharshi-gor replied:
This will not draw a plane, this is just to tell the the actor which plane to use to intersect a line. 

The tutorial itself creates a square actor to draw the plane. But the projection actor only shows the points.

Let me know if you need more information on this.

skoudoro reviewed (changes requested):
Hi @maharshi-gor,

I will continue my review, but here, 2 issues that needs to be fixed

maharshi-gor committed:
ENH: Added string options for basic planes.

- Added `XY`, `YZ` and `XZ` options to ommit the equation in the plane
  attribute.
